src/parser/jobs/ast/modules/ArcanaTracking/CritAndDHRates.tsx

```diff
@@ -0,0 +1,422 @@
+import {Action} from 'data/ACTIONS'
+import {Status} from 'data/STATUSES'
+import {Event, Events, SourceModifier} from 'event'
+import math from 'mathjsCustom'
+import {Analyser} from 'parser/core/Analyser'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Actor, Actors} from 'parser/core/modules/Actors'
+import {Data} from 'parser/core/modules/Data'
+
+const BASE_CRIT_PROBABILITY = 0.05 //5%
+const BASE_CRIT_MULTIPLIER = 0.4
+const ROUNDING_FACTOR = 0.001 //to match rounding used in game for crit rates
+
+export class CritAndDHPredictor extends Analyser {
+	static override handle = 'critAndDHQuery'
+
+	@dependency private data!: Data
+	@dependency private actors!: Actors
+
+	private playerControlled: Array<Actor['id']> = []
+	private pets: Array<Actor['id']> = []
+
+	//variable for output
+	private estCritDH: {[actor: Actor['id']]: {crit: number, DH: number}} = {}
+
+	//to initialize
+	private completeInitialized: boolean = false
+
+	//format for damage event
+	private damageEvent: Array<{actor: Actor['id'], timestamp: number, action: Status['id'] | Action['id'], critMod: number, DHMod: number, sourceModifier: number}> = []
+	private statusEvent: Array<{actor: Actor['id'], timestamp: number, action: Status['id'] | Action['id'], critRate: number, DHRate: number}> = []
+	private critPartyModifierCurrent: {[actor: Actor['id']]: {currentModifier: number}} = {}
+	private critEnemyModifierCurrent: {[actor: Actor['id']]: {currentModifier: number}} = {}
+	private DHPartyModifierCurrent: {[actor: Actor['id']]: {currentModifier: number}} = {}
+	private appliedStatuses: Array<{actor: Actor['id'], status: Status['id']}> = [] //reasoning is that bard's statuses are reapplied without removal. this is to catch those
+
+	//crit and DH Modifier variables
+	private critPartyModifiers: {[id: Status['id']]: {strength: number}} = {
+		[this.data.statuses.BATTLE_LITANY.id]: {strength: 0.1},
+		[this.data.statuses.DEVILMENT.id]: {strength: 0.2},
+		[this.data.statuses.THE_WANDERERS_MINUET.id]: {strength: 0.02},
+		[this.data.statuses.INNER_RELEASE.id]: {strength: 1},
```

Comment:
Inner Release only applies to Fell Cleave and Decimate, but this will buff all damage including autos.

Reply:
Thanks for that catch!
I've updated this to have a separate section along with Opo for MNK
See below commit